### PR TITLE
[Feature] Chat scrollback

### PR DESCRIPTION
**Pull request by DayBr3ak**
This PR changes the maxsize of the chatController buffer. Users can set their own size. Default is 200 (twitch default).

### CODE REVIEW

File: src/modules/chat_scrollback_settings/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+function updateSizeSettings() {
+    const sizeString = storage.get('chatScrollbackSize') || '';
+    let size;
+    if (sizeString.trim().length === 0) {
+        size = DEFAULT_SIZE;
+    } else {
+        try {
+            size = parseInt(sizeString, 10);
```
Comment (night) [inline src/modules/chat_scrollback_settings/index.js]: i dont think parseInt can throw

File: src/modules/chat_scrollback_settings/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const watcher = require('../../watcher');
+const storage = require('../../storage');
+const twitch = require('../../utils/twitch');
+
+const DEFAULT_SIZE = 200;
+const PROMPT_BODY = `Change the number of messages the chat is able to scrollback to. Increasing this value can slow down your browser...
+Leaving this value empty restore default settings.
```
Comment (night) [inline src/modules/chat_scrollback_settings/index.js]: this sentence is not proper grammar

File: src/modules/chat_scrollback_settings/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+const watcher = require('../../watcher');
+const storage = require('../../storage');
+const twitch = require('../../utils/twitch');
+
+const DEFAULT_SIZE = 200;
```
Comment (night) [inline src/modules/chat_scrollback_settings/index.js]: we should pull this value from twitch, not make our own

File: src/modules/chat_scrollback_settings/index.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+function changeSizeSettings(promptBody, storageID) {
+    const size = prompt(promptBody, storage.get(storageID) || DEFAULT_SIZE);
+    if (size !== null) {
+        storage.set(storageID, size.trim());
+    }
+}
+
+function updateSizeSettings() {
+    const sizeString = storage.get('chatScrollbackSize') || '';
```
Comment (night) [inline src/modules/chat_scrollback_settings/index.js]: we have a legacy key for this we should use: `scrollbackAmount`

File: src/modules/chat_scrollback_settings/index.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    let size;
+    if (sizeString.trim().length === 0) {
+        size = DEFAULT_SIZE;
+    } else {
+        try {
```
Comment (night) [inline src/modules/chat_scrollback_settings/index.js]: i dont get why the business logic of parsing is here. it should be coerced after the prompt to change, then stored as a number

File: src/modules/chat_scrollback_settings/index.js (hunk 1 above)
Comment (DayBr3ak) [reply src/modules/chat_scrollback_settings/index.js]: True... my bad

File: src/modules/chat_scrollback_settings/index.js (hunk 2 above)
Comment (DayBr3ak) [reply src/modules/chat_scrollback_settings/index.js]: I've changed it in my new commit. If it's still not good, help me please, english isn't my first language :)

**Comment by night** [comment]
So.. given your other pull request.. doesn't this get overwritten when you scroll?

**Comment by DayBr3ak** [comment]
Yes, but scrolling is temporary usually. It reverts back to the settings when the user resumes autoscrolling.

**Comment by night** [comment]
closing due to inactivity

**Pull request closed without merging**